domain-services/ifix-es-pipeline/src/main/java/org/egov/ifixespipeline/consumers/IfixElasticSearchPipelineListener.java

```diff
@@ -0,0 +1,116 @@
+package org.egov.ifixespipeline.consumers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.contract.request.RequestHeader;
+import org.egov.ifixespipeline.models.*;
+import org.egov.ifixespipeline.producer.Producer;
+import org.egov.ifixespipeline.repository.ServiceRequestRepository;
+import org.egov.ifixespipeline.service.FiscalDataEnrichmentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+@Service
+@Slf4j
+public class IfixElasticSearchPipelineListener {
+
+    @Autowired
+    private FiscalDataEnrichmentService fiscalDataEnrichmentService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ServiceRequestRepository serviceRequestRepository;
+
+    @Autowired
+    private Producer producer;
+
+    @Value("${fiscal.event.es.push.topic}")
+    private String indexFiscalEventsTopic;
+
+    @Value("${fiscal.event.kafka.push.topic}")
+    private String fiscalEventsNewRecordsTopic;
+
+    @Value("${ifix.master.data.service.host}")
+    private String ifixMasterDataServiceHost;
+
+    @Value("${ifix.master.data.service.search.endpoint}")
+    private String ifixMasterDataServiceSearchEndpoint;
+
+    @Value("${coa.electricity.head.name}")
+    private String electricityCoaHeadName;
+
+    @Value("${coa.operations.head.name}")
+    private String operationsCoaHeadName;
+
+    @Value("${coa.salary.head.name}")
+    private String salaryCoaHeadName;
+
+    private Map<String, Map<String, HashSet<String>>> tenantIdVsExpenditureTypeVsUuidsMap = new HashMap<>();
+
+    /**
+     * Kafka consumer
+     *
+     * @param record
+     * @param topic
+     */
+    @KafkaListener(topics = { "${fiscal.event.kafka.push.topic}", "${fiscal.event.migration.origin.push.topic}"})
+    public void listen(HashMap<String, Object> record, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        try {
+            FiscalEventRequest incomingData = objectMapper.convertValue(record, FiscalEventRequest.class);
+
+            // Enrich hierarchy map according to the tenantid encountered by this pipeline to avoid redundant network calls
+            if(!tenantIdVsExpenditureTypeVsUuidsMap.containsKey(incomingData.getFiscalEvent().getTenantId()))
+                tenantIdVsExpenditureTypeVsUuidsMap.put(incomingData.getFiscalEvent().getTenantId(), loadExpenditureTypeVsUuidMap(incomingData.getFiscalEvent().getTenantId()));
```

Comment:
You can write `loadExpenditureTypeVsUuidMap` method as `@PostConstruct`. Then you wouldn't need to write this logic in the listen method.

Reply:
we don't have the list of tenantIds at the time of loading, that is why we cannot add it to @PostConstruct method.